dbt/include/snowflake/macros/utils/cast.sql

```diff
@@ -0,0 +1,9 @@
+{% macro snowflake__cast(field, type) %}
+    {% if (type|upper == "GEOGRAPHY") -%}
+        to_geography({{field}})
+    {% elif (type|upper == "GEOMETRY") -%}
+        to_geometry({{field}})
+    {% else -%}
+        cast({{field}} as {{type}})
+    {% endif -%}
+{% endmacro %}
```

Comment:
## Backstory

### Commonality between unit testing and data contracts

dbt unit testing and dbt model contracts have a key commonality:
- both rely on being able to cast a null value to an explicit data type

Example in data contracts:
- `cast(null as integer) as my_column`

Example in unit testing:
- `try_cast(null as integer) as my_column`

### Bug in data contracts

Within https://github.com/dbt-labs/dbt-snowflake/issues/894, a user reported a bug related to casting `null as GEOGRAPHY` within data contracts.

### Solution

The [proposed path forward](snowflake/issues/894#issuecomment-1920281607) is to proactively avoid similar issues within unit testing _and_ fix the data contracts issue with these steps:

> 1. Within [dbt-adapters](https://github.com/dbt-labs/dbt-adapters), implement [#5916](https://github.com/dbt-labs/dbt-core/issues/5916) 
> 1. Within dbt-adapters, update [here](https://github.com/dbt-labs/dbt-adapters/blob/8f259f56e298c999337062589249f8673425f5db/dbt/include/global_project/macros/adapters/columns.sql#L56) to use it
> 1. Within dbt-snowflake, implement an override for [#5916](https://github.com/dbt-labs/dbt-core/issues/5916)
> 

(1) was addressed in [dbt-labs/dbt-adapters@5a50be7](https://github.com/dbt-labs/dbt-adapters/pull/55/commits/5a50be73163514dd7055af046c5223b092ee9b61)

(3) is addressed  by the code highlighted above.

(2) doesn't have a PR yet.

Reply:
@dbeatty10 To clarify, are you looking for an update on this PR? Or is this more to document the connection between the related issues?

Reply:
No update need on this PR -- it is perfect!

Just documenting the connection between the related issues. And partially so I could link it within https://github.com/dbt-labs/dbt-snowflake/issues/894#issuecomment-1936133733